### PR TITLE
fix: BGD Review fixes

### PR DESCRIPTION
**Pull request by 0xrajath**
## Motivation
BGD Review fixes

## Changes
1. USDC Cap has been removed. And a 100k BAL Cap has been added to bonding curve
2. Mechanism to deposit any remaining USDC in Aave V2 Collector back to aUSDC on behalf of the Collector after 100k BAL Cap has been filled.
3. Mechanism to deposit acquired BAL to aBAL in order to earn yield after 100k BAL Cap has been filled.
4. Emergency function to rescue any tokens that have been accidentally sent to Bonding Curve contract by sending them to Aave V2 Collector.
5. Removing legacy chainlink validation.
6. Revised precision math implementation

## Reference:
1. BGD review comments: https://gist.github.com/sakulstra/485a218d65a78e62d6edde7da11e3f95#bal-bonding-curve-review
7. Chainlink validation: https://ethereum.stackexchange.com/questions/133890/chainlink-latestrounddata-security-fresh-data-check-usage/133897#133897

### CODE REVIEW

File: src/OneWayBondingCurve.sol [hunk 1]
```diff
@@ -87,49 +79,75 @@ contract OneWayBondingCurve {
...
-    function getBondingCurvePriceMultiplier() public pure returns (uint256) {
-        return
-            FixedPointMathLib.mulDivDown(
-                BASIS_POINTS_GRANULARITY + BASIS_POINTS_ARBITRAGE_INCENTIVE,
-                USDC_BASE,
-                BASIS_POINTS_GRANULARITY
-            );
+    /// @notice Transfer any tokens accidentally sent to this contract to Aave V2 Collector
+    /// @param tokens List of token addresses
+    function rescueTokens(address[] memory tokens) external {
```
Comment (AustinGreen) [inline src/OneWayBondingCurve.sol:148]: why can't this be `address[] calldata tokens`

**Review by AustinGreen — APPROVED** [review]
LGTM just had one small question

File: src/OneWayBondingCurve.sol (hunk 1 above)
Comment (0xrajath) [reply src/OneWayBondingCurve.sol:148]: Oh good question. I didn't think of it. Let me try it out.

File: src/ProposalPayload.sol [hunk 2]
```diff
@@ -46,26 +47,31 @@ contract ProposalPayload {
...
             address(oneWayBondingCurve),
             usdcAmount
         );
+
+        // 4. Approve the One Way Bonding Curve contract to spend 100K BAL from AAVE V2 Collector
+        IAaveEcosystemReserveController(AaveV2Ethereum.COLLECTOR_CONTROLLER).approve(
+            AaveV2Ethereum.COLLECTOR,
+            BAL_TOKEN,
+            address(oneWayBondingCurve),
+            BAL_AMOUNT_CAP
```
[the remark this answers is not in the document]
Comment (0xrajath) [reply src/ProposalPayload.sol:74]: Yup. Forgot about that. Will change it to 300k allowance.

File: src/OneWayBondingCurve.sol [hunk 3]
```diff
@@ -87,49 +79,75 @@ contract OneWayBondingCurve {
...
-    /// @return amountOut the amount of USDC received
-    function getAmountOut(uint256 amountIn) public view returns (uint256 amountOut) {
-        // Normalizing input BAL amount from BAL decimals (18) to USDC decimals (6)
-        uint256 normalizedAmountIn = normalizeFromBALDecimalsToUSDCDecimals(amountIn);
-        // The actual USDC value of the input BAL amount
-        uint256 usdcValueOfAmountIn = FixedPointMathLib.mulDivDown(getOraclePrice(), normalizedAmountIn, USDC_BASE);
-        // The incentivized USDC value of the input BAL amount
-        amountOut = FixedPointMathLib.mulDivDown(getBondingCurvePriceMultiplier(), usdcValueOfAmountIn, USDC_BASE);
+    /// @return amountOutWithBonus the amount of USDC received with 50 bps incentive included
+    function getAmountOut(uint256 amountIn) public view returns (uint256 amountOutWithBonus) {
```
[the remark this answers is not in the document]
Comment (0xrajath) [reply src/OneWayBondingCurve.sol]: Yup. Done.

File: src/OneWayBondingCurve.sol [hunk 4]
```diff
@@ -87,49 +79,75 @@ contract OneWayBondingCurve {
...
 
-    /// @notice Normalize BAL/USD Chainlink Oracle Feed decimals to USDC decimals (6)
-    function normalizeFromOracleDecimalstoUSDCDecimals(uint256 amount) public view returns (uint256) {
-        return FixedPointMathLib.mulDivDown(amount, USDC_BASE, 10**uint256(BAL_USD_FEED.decimals()));
+    /// @notice Deposit all acquired BAL after 100k BAL Amount Cap has been filled
+    function depositBalCollector() external {
+        uint256 balBalance = BAL.balanceOf(AaveV2Ethereum.COLLECTOR);
+        uint256 balAllowance = BAL.allowance(AaveV2Ethereum.COLLECTOR, address(this));
+
+        if (totalBalReceived < BAL_AMOUNT_CAP) revert BalCapNotFilled();
```
[the remark this answers is not in the document]
Comment (0xrajath) [reply src/OneWayBondingCurve.sol]: Done

File: src/ProposalPayload.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xrajath) [reply src/ProposalPayload.sol]: Done

**Pull request merged**